martin/src/pg/table_source.rs

```diff
@@ -19,7 +19,11 @@ static DEFAULT_EXTENT: u32 = 4096;
 static DEFAULT_BUFFER: u32 = 64;
 static DEFAULT_CLIP_GEOM: bool = true;
 
-pub async fn query_available_tables(pool: &PgPool) -> Result<SqlTableInfoMapMapMap> {
+pub async fn query_available_tables(
+    pool: &PgPool,
+    clip_geom: Option<bool>,
```

Comment:
I am not sure, but it might be better to actually change `buffer` and `clip_geom` here to `None` to indicate that we don't have them yet, and keep them unset until much later as part of the `instantiate_tables` somewhere around `db_inf.srid = srid;` line.

Reply:
Like:
```rust
...
db_inf.srid = srid;
update_id_column(&id2, &mut db_inf, auto_tables);
impose_clip_geom(&mut db_inf, auto_tables);
impose_buffer(&mut db_inf, auto_tables);
...
```

Reply:
sure, just maybe it will make sense to rename the `update_id_column` to `update_auto_fields` and do all these steps inside one function

Reply:
Have a good day @nyurik It's still a draft, but could you help me a code review?